### ISSUE TITLE
scrolling direction when using inverted prop

### ISSUE THREAD

**Issue by tafelito** [necolas/react-native-web#995]
**The problem**
When using FlatList with inverted prop, the mouse scrolling direction is also inverted. I'm not sure if this is an issue per se, but I think only the content of the list should be inverted and not the direction of the scroll. I worked on implementation for the same feature for the [RLV](https://github.com/Flipkart/recyclerlistview) and I tried to handle this on the onScroll event reversing the scroll direction. This only happens on desktops, mobiles scroll is different so it works fine there

**How to reproduce**
Just using the FlatList with inverted={true}

**Expected behavior**
Invert list content but keep mouse scrolling direction unchanged

**Environment (include versions). Did this work in previous versions?**

* OS:
* Device:
* Browser: Chrome, Safari
* React Native for Web (version):^0.8.3
* React (version): 16.4

**Comment by necolas** [comment]
> Is this something that could be done here as well?

Sounds like a good idea

**Comment by necolas** [comment]
I'm open to a PR if you feel like it. No worries if not. Off the top of my head, I'm not sure where the best place for this to go is but can help figure that out if you're going to try

**Comment by tafelito** [comment]
I'd be happy to try. I have to figure it out how to run RNW locally first and then any help would be great

**Comment by necolas** [comment]
Cool. This setup guide will get you started https://github.com/necolas/react-native-web/blob/master/.github/CONTRIBUTING.md

**Comment by tafelito** [comment]
So I saw that I can add the onWheel prop to the scrollProps in the VirtualizedList and I could call `this._scrollRef.scrollTo` setting y or x to the delta but it doesn't work.
I also tried a different approach but didn't work either. Instead of scaling the elements with -1, I used flexDirection to reverse the scrollview entirely but even though the items are reversed, they are still mounted in the same order, so you'll see how the items are pushed down while loading. So it's not a solution.

**Comment by mglagola** [comment]
@tafelito did you ever find a solution?

**Comment by tafelito** [comment]
I haven't had more time to spend on this unfortunately neither or the useWindow scroll feature

**Comment by tafelito** [comment]
@necolas are you using FlatList at Twitter Lite?  Because I saw that the messages window uses an inverted scroll and the direction works fine there. 

Also I noticed that the home page uses some sort of window scroller but on the chat screen you can only scroll on the list and not outside

**Comment by necolas** [comment]
No. Twitter PWA uses a custom scroller that isn't integrated with React Native at all, i.e., no support for the responder system etc

**Comment by danalloway** [comment]
did some work on this over in #1241

**Comment by unadlib** [comment]
I've had the same issue, and I'm not sure it doesn't seem to have a fix plan?

**Comment by raphaelrk** [comment]
Here's a version of that same codesandbox example with some scrolljacking to invert the scroll direction and a trick to force hardware acceleration for better scroll performance: [https://codesandbox.io/s/react-native-h32if](https://codesandbox.io/s/react-native-h32if)

Tested on macOS Mojave FF/Safari/Chrome July 14 2019

**Comment by shaylew** [comment]
There's also arrow keys, page up/page down, home/end, and mousewheel drag to consider. I think I've come up with a proof of concept for simpler way to do this that doesn't require reimplementing each type of scrolling: https://jsfiddle.net/4t2uyfpz/1/

The idea is to keep two scrollable elements: a wrapper with a visible scrollbar for the user, and an inverted inner element with `position: sticky` that gets scrolled by the script. Scroll events -- mousewheel or keyboard -- move the wrapper's scrollbar and we scroll the inner content accordingly. If items get added or removed inside the inverted list we scroll and resize the wrapper to keep it in sync.

(Right now there's some weird behavior where if you scroll to the top and then remove N elements, Chrome -- and only Chrome -- keeps you stuck to the top until you either scroll manually or add N+1 elements back. I'm not sure exactly what's up with that but there's hopefully a workaround.)

**Comment by davidfant** [comment]
Any update on this?

**Comment by vujevits** [comment]
one thing to note is that on Firefox the scroll direction works correctly.

**Comment by divonelnc** [comment]
Is there any progress on this issue?

**Comment by smithydll** [comment]
@raphaelrk I had to add flatlistRef.current to the useEffect dependency otherwise the content in the flatlist would fire the effect several times

```
useEffect(() => {

[snip]

}, [flatlistRef.current]); // needs to run any time flatlist mounts
```

**Comment by roryabraham** [comment]
Heads up @staltz and @necolas – we've been using [a workaround](https://github.com/Expensify/App/blob/adb0d8a52fc7210e8e48e4439f83d932a064e1f6/src/components/InvertedFlatList/index.js) for the inverted wheel event thing for a while now. I saw [this commit](https://github.com/necolas/react-native-web/commit/7ec2489c78f9734bae5bcfdc514b0c91621a6300) is in the pipeline to fix this issue, and it's more or less the same as the workaround we've been using.

However, we did have someone report an [edge case](https://github.com/Expensify/App/issues/8612) with this solution though – if you use `Tab` to focus the scrollbar, then pressing `ArrowDown` scrolls you up and `ArrowUp` scrolls you down. Maybe worth fixing for accessibility reasons?

**Comment by staltz** [comment]
@roryabraham We solved the keyboard problem like this: https://github.com/staltz/manyverse/blob/master/patches/react-native-web%2B0.17.6.patch (I think it's too ugly to make an upstream PR)
